### ISSUE TITLE
mysql.initialDatabases.….schema is not of type `null or path`

### ISSUE THREAD

**Issue by schneefux** [NixOS/nixpkgs#58582]
## Issue description
I have the following mysql configuration:
```nix
services.mysql = {
  enable = true;
  package = pkgs.mysql;
  initialDatabases = [
    { name = "nextcloud"; schema = ""; }
  ];
  initialScript = /etc/nixos/setup-credentials.sql;
};
```
With the changes from #57550, I get the error
> The option value 'services.mysql.initialDatabases.[definition 1-entry 1].schema" in `/etc/nixos/configuration.nix' is not of type 'null or path'.

Is this intentional?

I looked further and saw that as of #25950, `schema` is now optional. Removing it I get the error
> cannot coerce null to a string, at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/databases/mysql.nix:363:66

### Steps to reproduce

see above

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.98, NixOS, 19.09pre173381.2b9aa0ac90e (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.09pre174075.aaae785f903"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by schneefux** [comment]
cc @florianjacob

**Comment by florianjacob** [comment]
@schneefux what's your intention with `schema = "";`? If you want to create an empty database, the intention is to use `schema = null;`. As this is the default, the idea is indeed to use
```
initialDatabases = [
    { name = "nextcloud"; }
  ];
```
but as there is a default value now, the previous check “is that attribute defined” broke.

The `schema = "";` previously worked because the creation script silently falls through if the schema file does not exist, which is bad. I'll create a PR to document this, and to fix the bug.

**Comment by schneefux** [comment]
fixed with #58702
